### PR TITLE
Empty out search.html; perform AJAX request instead.

### PR DESCRIPTION
**Pull request by pelmers**
Addresses bug [970405](https://bugzilla.mozilla.org/show_bug.cgi?id=970405).
~~Still a bit rough; sometimes when loading a page the location will change after about half a second.~~ Fixed by #423.

Here we remove the content block of search.html and instead perform an AJAX request on load of the search page based on the search portion of the location.
Remark: I do not completely remove search.html because I want to preserve the back button and the ability to link directly to some search results.

### CODE REVIEW

**Comment by pelmers** [comment]
Strange behavior: if you close a tab, and then re-open it, then the page shows a big JSON dump. Hitting refresh solves the issue. This occurs in Chrome and Firefox.

**Comment by erikrose** [comment]
That is weird. IHNI. It happens both for Return-key-pressed results and non.

**Comment by erikrose** [comment]
Unsurprisingly, when you hit Return, there's a little flash of blank-search-results before the AJAX pulls them in. It's trivial locally, but I worry it'll be more noticeable on real servers. Why don't we intercept the Return-key press rather than actually submitting the form and just substitute the results onto the existing page?

**Comment by pelmers** [comment]
If we intercept the form we will still have a delay before the new results are shown. In fact I think it would be pretty much the same delay, since it would be sending the same request. Do you think it would just be less noticeable?

**Comment by erikrose** [comment]
The difference would be that the search bar itself wouldn't flash: there wouldn't be that appearance of a reload without actually being complete. And we can and should activate the spinner while the AJAX request is in flight.

**Comment by erikrose** [comment]
Obviously, we'll need some special code to handle redirects, but it should be trivial.

**Comment by pelmers** [comment]
Updated, I made the change by responding with 'redirect: {location}' when a direct hit is found.

**Comment by erikrose** [comment]
The docstrings for _search_json() and _search_html() need to be updated.

File: dxr/static/js/dxr.js [hunk 1]
```diff
@@ -423,19 +408,32 @@ $(function() {
...
+                if (data.redirect) {
+                    window.location.href = data.redirect;
+                    return;
+                }
+                data.query = query;
+                // New results, overwrite
+                if (myRequestNumber > displayedRequestNumber) {
+                    displayedRequestNumber = myRequestNumber;
+                    populateResults(data, false);
+                    historyWaiter = setTimeout(history.pushState.bind(history, {}, '', queryString),
```
Comment (erikrose) [inline dxr/static/js/dxr.js]: If the user hit Return (i.e. redirect is true), let's update the URL immediately instead of waiting.

**Comment by erikrose** [comment]
Otherwise, looks really good. This'll make our code simpler with a break-even effect on UX (piecewise load for hits to search.html from scratch but no more page flash when hitting Return in the search field).

**Comment by erikrose** [comment]
r+ after you make that last tweak. No need to go through another round.

**Pull request merged**